src/engine/controls/vinylcontrolcontrol.cpp

```diff
@@ -89,25 +89,34 @@ void VinylControlControl::slotControlVinylSeek(double fractionalPos) {
         return;
     }
 
-    double total_samples = getSampleOfTrack().total;
-    double new_playpos = round(fractionalPos * total_samples);
+    const auto trackEndPosition =
+            mixxx::audio::FramePos::fromEngineSamplePosMaybeInvalid(
+                    getSampleOfTrack().total);
+    if (!trackEndPosition.isValid()) {
+        return;
+    }
+    const auto newPlayPos = trackEndPosition * fractionalPos;
```

Comment:
`newPlayPos = kStartFramePos + (trackEndPosition - kStartFramePos) * fractionalPos;`?

I suppose there are much more spots where positions and offsets/distances are used confused. Those could only be detected and fixed by introducing a distinct type for `FrameDiff_t`.

Reply:
5dc279b9c872ad61e74042cc9b8617e4103a431d